native/core/src/execution/jni_api.rs

```diff
@@ -508,9 +505,6 @@ pub unsafe extern "system" fn Java_org_apache_comet_Native_executePlan(
             let next_item = exec_context.stream.as_mut().unwrap().next();
             let poll_output = exec_context.runtime.block_on(async { poll!(next_item) });
 
-            // Update metrics
-            update_metrics(&mut env, exec_context)?;
```

Comment:
I wonder if we should add a config so that we can choose between frequent metrics updates vs just updating once the query completes. It can sometimes be helpful to see live metrics.

Reply:
I do like the idea of updating metrics every N seconds

Reply:
I think checking a coarse-grained clock (i.e., `CLOCK_MONOTONIC_COARSE`) to see if N seconds has elapsed to produce updated metrics would be a reasonable compromise on performance impact vs. fresh metrics.

Reply:
I also like the idea of updating every N seconds. One good reason for updating frequently is to keep updating the live UI.

Reply:
@mbutrovich  Thank you for your idea, sounds great to me, I will try to do that later.